src/System.Text.Json/tests/Serialization/Value.ReadTests.GenericCollections.cs

```diff
@@ -1154,5 +1155,81 @@ public static void ReadReadOnlyCollections_Throws()
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<ReadOnlyStringICollectionWrapper>(@"[""1"", ""2""]"));
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<ReadOnlyStringToStringIDictionaryWrapper>(@"{""Key"":""key"",""Value"":""value""}"));
         }
+
+        [Fact]
+        public static void HigherOrderCollectionInheritance_Works()
+        {
+            static void RunTest<T>(T instance)
+            {
+                string JSON;
```

Comment:
nit: use camel casing

To avoid naming conflict, maybe use expectedJson
```suggestion
                string expectedJson;
```